robot/web/web.go

```diff
@@ -627,6 +627,12 @@ func (svc *webService) startStream(streamFunc func(opts *webstream.BackoffTuning
 func (svc *webService) startImageStream(ctx context.Context, source gostream.VideoSource, stream gostream.Stream) {
 	ctxWithJPEGHint := gostream.WithMIMETypeHint(ctx, rutils.WithLazyMIMEType(rutils.MimeTypeJPEG))
 	svc.startStream(func(opts *webstream.BackoffTuningOptions) error {
+		if cam, ok := source.(camera.Camera); ok {
+			src, err := camera.SourceFromCamera(cam)
+			if err == nil {
```

Comment:
`if err != nil` that means for some reason we can't get a source from our camera. I'm not sure if that should be an error but it should at least be a warning so we aren't ignoring potential errors. That is
```
if cam, ok := source.(camera.Camera); ok {
    src, err := camera.SourceFromCamera(cam)
    if err != nil {
        warn(err, ...)	// or return error
        break // do not accidentally assign source = nil 	
    }
    source = src
}

```